server/pull_request.go

```diff
@@ -113,27 +113,27 @@ func (s *Server) handlePullRequestEvent(event *PullRequestEvent) {
 
 			mlog.Info("test server instance", mlog.String("test server", spinmint.InstanceID))
 			mlog.Info("Will destroy the test server for a merged/closed PR.")
-			s.sendGitHubComment(pr.RepoOwner, pr.RepoName, pr.Number, s.Config.DestroyedSpinmintMessage)
+			s.sendGitHubComment(ctx, pr.RepoOwner, pr.RepoName, pr.Number, s.Config.DestroyedSpinmintMessage)
 			go s.destroySpinmint(pr, spinmint.InstanceID)
```

Comment:
We should also create a context for the spinmint destruction, in the AWS API calls.

Reply:
You mean check for `ctx.Done()` in the `destroySpinmint` method right? AWS API calls don't accept context

Reply:
I meant the AWS API calls. `destroySpinmint` happens in it's own goroutine so that need not take the same context.

From what I can see every AWS API has a `WithContext` counterpart. For example: https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.RunInstancesWithContext.

used in server/spinmint.go:L150.

Is there any API in particular that doesn't take context?

Reply:
Ok, I misunderstood the point. :+1:

Reply:
Added